app/models/recipe.rb

```diff
@@ -26,6 +27,58 @@ class Recipe < ActiveRecord::Base
     greater_than_or_equal_to: 0
   }
 
+  def self.recommend(user)
+    favorites = []
+    high_favorites = []
+    good_ingredients = Hash.new(0)
+    top_suggestions = []
+
+    user.favorites.each do |favorite|
+      favorites.push(favorite.recipe)
+      rating = Rate.find_by(rater_id: user, dimension: "value", rateable_id: favorite.recipe)
+      high_favorites.push favorite.recipe if rating >= 4 || rating.nil?
+    end
+
+    high_favorites.each do |recipe|
+      recipe.ingredients.each do |ingredient|
+        rating = Rate.find_by(rater_id: user, dimension: "value", rateable_id: favorite.recipe)
+
+        # build an scored hash of favored ingredient keywords
+        ingredient.keywords.each do |keyword|
+          if rating == 5
+            good_ingredients[keyword.noun] += 5
+          elsif rating == 4
+            good_ingredients[keyword.noun] += 3
+          else
+            good_ingredients[keyword.noun] += 2
+          end
+        end
+      end
+    end
+
+    # lets score some recipe suggestions
+    suggestions = Hash.new(0)
+
+    search_keys = (good_ingredients.sort_by {|k, v| -v})
+    search_keys[0..9].each do |keyword|
+      keyword[0].ingredients.each do |ingredient|
+        ingredient.recipes.each do |recipe|
+          suggestions[recipe] += good_ingredients[keyword]
+        end
+      end
+    end
+
+    # find the best recipes
+    suggestions.sort_by {|k, v| -v}.each do |ranked_suggestion|
```

Comment:
Unused block argument - `k`. If it's necessary, use `_` or `_k` as an argument name to indicate that it won't be used.<br>Space between { and | missing.<br>Space missing inside }.